internal/k8shandler/cluster.go

```diff
@@ -400,3 +405,37 @@ func addNodeState(node NodeTypeInterface, nodeStatus *api.ElasticsearchNodeStatu
 	nodeStatus.DeploymentName = nodeState.DeploymentName
 	nodeStatus.StatefulSetName = nodeState.StatefulSetName
 }
+
+func (er *ElasticsearchRequest) checkWatermarkAndUnblockIndices() {
+	er.refreshDiskWatermarkThresholds()
+	if er.isDiskUtilizationBelowFloodWatermark() {
+		indices, err := er.esClient.GetAllIndices("")
+		if err != nil {
+			log.Error(err, "failed to fetch all indices")
+		}
+		for _, index := range indices {
+			if index.Index == constants.SecurityIndex {
+				continue
+			}
+			if er.isIndexBlocked(index.Index) {
+				if err := er.unblockIndex(index.Index); err != nil {
+					log.Error(err, "Couldn't update the index setting")
+				}
+			}
+		}
+	}
+}
+
+func (er *ElasticsearchRequest) isDiskUtilizationBelowFloodWatermark() bool {
```

Comment:
nit: i would expect that if we were testing this function it would be in `cluster_test.go`

Reply:
@ewolinetz updated the file.